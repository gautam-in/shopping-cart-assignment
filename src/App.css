body{
    font-family: 'Dosis', sans-serif;;
    /* padding: 20px 60px; */
}

a {
    text-decoration: none;
    color: black;
    cursor: pointer;
}

* {
    box-sizing: border-box;
}

