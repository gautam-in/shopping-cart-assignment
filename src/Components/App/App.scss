@import url("https://fonts.googleapis.com/css?family=Dosis:400,600,700");
.copyright {
  text-align: center;
  padding: 0.5rem;
  background: #0000000d;
}
* {
  font-family: "Dosis";
}
