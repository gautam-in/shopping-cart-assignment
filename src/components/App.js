import React, { Component } from 'react';
import { connect } from 'react-redux';
import {withRouter} from 'react-router';
import toastr from 'toastr';
import { Modal } from "react-responsive-modal";
import {resetCartData} from '../actions/productActions';
import Constants from '../constants';
import Cart from '../presentations/Cart';
import Header from '../presentations/Header';
import Footer from '../presentations/Footer';
class App extends Component {
    constructor(props) {
        super(props);
        this.state = {openModal:false}
    }
    componentDidMount(){
        window.addEventListener('offline', this.checkBroswerState);
        window.addEventListener('online', this.checkBroswerState);
    }
    componentWillUnmount(){
        window.removeEventListener('offline',this.checkBroswerState);
        window.removeEventListener('online',this.checkBroswerState);
    }
    checkBroswerState=()=>{
        if(navigator.onLine == true){
            this.online()
        } else if(navigator.onLine == false){
            this.offline();
        }
    }
    checkoutComplete = () => {
        this.props.resetCartData();
        this.onCloseModal();
        toastr.success(Constants.TEXTS.TOASTS.cart_emptied,Constants.TEXTS.TOASTS.order_placed,{timeOut:1000});
    }
    showCartView=()=>{
        this.setState({openModal: true});
        document.getElementsByTagName('body')[0].style.overflow = 'hidden';
        document.getElementsByTagName('html')[0].style.overflow = 'hidden';
    }
    onCloseModal = () => {
        this.setState({ openModal: false });
        document.getElementsByTagName('body')[0].style.overflow = 'auto';
        document.getElementsByTagName('html')[0].style.overflow = 'auto';
      };
    offline=() => {
        toastr.warning('',Constants.TEXTS.DEFAULTS.offline_msg); 
    }
    online=()=>{
        toastr.success('',Constants.TEXTS.DEFAULTS.online_msg);
    }
    render(){
        const {productInfo,router} = this.props;
        const {openModal} = this.state;
        return (<React.Fragment>
            <Header router={router} cartClick={this.showCartView} cartInfo={productInfo.cartItems}></Header>
            {this.props.children}
            <Footer />
            {openModal ? (<Modal
                open={openModal}
                onClose={this.onCloseModal}
                center
                aria-labelledby={`My Cart(${productInfo.cartItems.legth} items)`}
                aria-describedby={Constants.TEXTS.MODAL.modal_desc}
                classNames={{
                    overlayAnimationIn: '',
                    overlayAnimationOut: '',
                    modalAnimationIn: Constants.TEXTS.MODAL.customEnterModalAnimation,
                    modalAnimationOut: Constants.TEXTS.MODAL.customLeaveModalAnimation
                }}
                animationDuration={800}
                showCloseIcon={false}>
                <Cart cartData={productInfo.cartItems} checkoutComplete={this.checkoutComplete} closeModal={this.onCloseModal}></Cart>
            </Modal>) : null}
        </React.Fragment>)
    }
}
function mapStateToProps(state){
    return{
        productInfo: state.productReducer
    }
}
const mapDispatchToProps=(dispatch,props)=>{
    return{
        resetCartData: ()=>{dispatch(resetCartData())}
    }
}
export default connect(mapStateToProps,mapDispatchToProps)(App);