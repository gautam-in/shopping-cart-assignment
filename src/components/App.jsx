import React, { Component, useEffect, useState } from 'react';
import {useSelector} from 'react-redux'
import { Route, Switch, BrowserRouter as Router, Redirect } from  'react-router-dom'

import * as Endpoints from './Endpoints'
import Header from './shared/Header'
import Footer from "./shared/Footer"
import Home from './pages/Home'
import ProductsPage from './pages/ProductsPage'
import LoginPage from './pages/LoginPage'
import RegisterPage from './pages/RegisterPage'
import Error404 from './pages/Error404'

function App(){
    const [menuOpen, setMenuOpen] = useState(false)
    const [cart, setCart] = useState(false);
    const [categories, setCategories] = useState([]);
    const this_cart = useSelector((store) => store.cart);
    localStorage.setItem("cart", JSON.stringify(this_cart));

    React.useEffect(()=>{
        var axios = require("axios");
        var configurations = {
          method: "get",
          url: Endpoints.get_categories,
          headers: {}
        };

        axios(configurations)
        .then((res)=>{
          
          //sort by order & filter out disabled categories
          let activeCategories = res.data.sort(function(a, b){
              return a.order - b.order;
          }).filter(a => a.enabled == true);

          setCategories(activeCategories);
        })
        .catch((err)=>{
          console.log(err);
        })
    }, []);

    return (
      <>
        <Router>
          <Header setCart={setCart} cart={cart}  menuOpen={menuOpen} setMenuOpen={() => setMenuOpen(!menuOpen)} />
          <main>
          <Switch>
                <Route exact path='/'>
                    <Redirect to="/home" />
                </Route>
                <Route exact path='/home'>
                    <Home categories={categories} />
                </Route>
                <Route exact path='/products'>
                    <ProductsPage categories={categories} setCart={setCart} />
                </Route>
                <Route exact path='/signin' component={LoginPage} />
                <Route exact path='/signup' component={RegisterPage} />
                <Route component={Error404} />
                
          </Switch>
          </main>
          
          <Footer />
        </Router>
      </>
    )
}

export default App;