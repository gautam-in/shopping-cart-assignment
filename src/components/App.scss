@import "~bootstrap/scss/bootstrap";
@import url("https://fonts.googleapis.com/css?family=Dosis:400,600,700");

$base-font-family: 'Dosis';
$base-color: #ffffff;
$base-font-size: 14px;
$medium-font-size : 16px;
$btn-font-color: #ffffff;
$btn-background-color: #D10057;
$border-dark: rgba($base-color, 0.88);
$menu-shadow: 0px 2px 10px 2px rgb(0 0 0 / 50%);
$category-shadow: 0px 15px 10px -15px rgb(0 0 0 / 50%);
$noitem-font-size: 18px;

#shopping-cart-app {
    .default-btn-style{
        border-radius: 0;
        border: none;
        color: $btn-font-color;
        background-color: $btn-background-color;
    }

    .heading-text {
        font-size: 32px;
        font-weight: 700;
        margin-bottom: 10px;
    }

    .category-shadow{
        box-shadow: $category-shadow;
    }
    
    .form-control:focus{
        outline: none;
        box-shadow: none;
    }

    .form-group {
        position: relative;
        padding: 15px 0 0;
        margin-top: 10px;
    }

    .form-field {
        font-family: inherit;
        width: 100%;
        border: 0;
        border-bottom: 1px solid #d2d2d2;
        outline: 0;
        font-size: 16px;
        color: #212121;
        padding: 7px 0;
        background: transparent;
        transition: border-color 0.2s;
        border-radius: 0;
    }

    .form-field::placeholder {
        color: transparent;
    }

    .form-field:placeholder-shown ~ .form-label {
        font-size: 16px;
        cursor: text;
        top: 20px;
    }

    label,
    .form-field:focus ~ .form-label {
        position: absolute;
        top: 0;
        display: block;
        transition: 0.2s;
        font-size: 12px;
        color: #9b9b9b;
    }

    .form-field:focus ~ .form-label {
        color: #6BCBD3;
    }

    .form-field:focus {
        padding-bottom: 6px;
        border-bottom: 2px solid #6BCBD3;
    }
}