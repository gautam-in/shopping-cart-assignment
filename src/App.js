import { Routes, Route } from 'react-router-dom';
import './App.css';
import Home from './pages/Home';
import 'bootstrap/dist/css/bootstrap.min.css';
import TopNavbar from './navigation/TopNavbar';
import Plpage from './pages/Plpage';
import RegisterForm from './components/register/RegisterForm'
import LoginComp from './components/login/LoginComp';
import FooterComp from './components/footer/FooterComp';

function App() {
  return (
    <>
    <Routes>
      {/* <Route path="/" element={<TopNavbar/>}> */}
        <Route index element={<Home />} />
        <Route path='plp' element={<Plpage />} />
        <Route path='signup' element={<RegisterForm />} />
        <Route path='login' element={<LoginComp />} />
      {/* </Route> */}

      
    </Routes>
    <FooterComp />
    </>
    
    
  );
}

export default App;
