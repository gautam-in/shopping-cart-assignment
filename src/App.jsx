import React, { lazy, Suspense } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./App.scss";
import ErrorBoundary from "./Component/ErrorBoundary/errorBoundary.component";
import Footer from "./Component/Footer/footer.component";
import Header from "./Component/Header/header.component";
import Spinner from "./Component/Spinner/spinner.component";
import CartContainer from "./Screen/Cart/cart.container";

const ProductPage = lazy(() => import("./Screen/Product/product.container"));
const LoginPage = lazy(() => import("./Screen/Login/login.container"));
const HomePage = lazy(() => import("./Screen/Home/home.container"));
const RegisterPage = lazy(() => import("./Screen/Register/register.conatiner"));

function App() {
  return (
    <Router>
      <div className="app-container">
        <Header />
        <Switch>
          <ErrorBoundary>
            <Suspense fallback={<Spinner />}>
              <Route path="/" exact component={HomePage} />
              <Route path="/product" component={ProductPage} />
              <Route path="/login" exact component={LoginPage} />
              <Route path="/register" exact component={RegisterPage} />
            </Suspense>
          </ErrorBoundary>
        </Switch>
        <CartContainer />
        <Footer />
      </div>
    </Router>
  );
}

export default App;
