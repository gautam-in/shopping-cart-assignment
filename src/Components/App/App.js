import React, { Suspense } from "react";
import { Provider } from "react-redux";
import { store } from "../../redux/store";
import Navbar from "../Navbar/Navbar";
const Home = React.lazy(() => import("../Home"));
const Cart = React.lazy(() => import("../Cart/Cart"));
const Productlandingpage = React.lazy(() =>
  import("../Productlandingpage/Productlandingpage")
);
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
const Login = React.lazy(() => import("../Login/Login"));
const Register = React.lazy(() => import("../Register"));
import "./App.scss";
export default function App() {
  return (
    <Provider store={store}>
      <Router>
        <div>
          <Navbar />
          <Suspense fallback={<div>Loading...</div>}>
            <Switch>
              <Route path="/" exact component={Home} />
              <Route path="/Login" component={Login} />
              <Route path="/Register" component={Register} />
              <Route path="/Cart" component={Cart} />
              <Route path="/Product" component={Productlandingpage} />
            </Switch>
          </Suspense>
          <Suspense fallback={<div>Loading...</div>}>
            <Cart />
          </Suspense>
          <footer className="copyright">
            <small>
              Copyright @ 2011-2018 Sabka Bazaar Grocery Supplies Pvt Ltd
            </small>
          </footer>
        </div>
      </Router>
    </Provider>
  );
}
