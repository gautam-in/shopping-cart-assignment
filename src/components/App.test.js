import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import App from "./App";

const mockStore = configureMockStore([thunk]);
const setUp = (initialState = { products: [] }) => {
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
          <App />
      </Provider>
    );
    return wrapper;
} 

describe("App component", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = setUp();
    });
    it("Should render without errors", () => {
        const component = wrapper.find("App");
        expect(component.length).toBe(1);
    });
})
