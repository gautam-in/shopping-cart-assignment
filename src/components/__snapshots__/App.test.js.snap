// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App Correctly 1`] = `ShallowWrapper {}`;
